client/src/details/connection/CurlConnection.cpp

```diff
@@ -424,3 +422,11 @@ void CurlConnection::ProcessRetry(int attempt, const Result& httpResult)
     LOG_INFO(m_handler, "Sleeping for %lld ms", static_cast<long long>(retryDelay.count()));
     std::this_thread::sleep_for(retryDelay);
 }
+
+void CurlConnection::BuildAndSetHeaders(CurlHeaderList& headers, const std::string& cv)

```

Comment:
defensive coding: what happens if someone calls this multiple times?

wouldn't it be better to have a CurlHeaderList on the stack here and return it as return value?